### ISSUE TITLE
Error while installing v1.2.2: checksum list has unexpected SHA-256 hash

### ISSUE THREAD

**Issue by ribbybibby** [community-terraform-providers/terraform-provider-ignition#13]
I am receiving an error about a checksum mismatch when trying to install `v1.2.2`.

### Terraform Version
```
$ terraform version
Terraform v0.14.2
```

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    ignition = {
      source  = "community-terraform-providers/ignition"
      version = "1.2.2"
    }
  }
}
```

### Debug Output
```
$ terraform init .

Initializing the backend...

Initializing provider plugins...
- Finding community-terraform-providers/ignition versions matching "1.2.2"...
- Installing community-terraform-providers/ignition v1.2.2...

Error: Failed to install provider

Error while installing community-terraform-providers/ignition v1.2.2: checksum
list has unexpected SHA-256 hash
b14ff03a705a08e92ad9d15f65f01f93ecd307b6a1190ec5c11a576bc6e2efb1 (expected
02e7eaa2691e20b476754b607abba6f191e175b9c13355b31d9c88b623b52cc1)
```

### References
- #10

**Comment by yussufsh** [comment]
I had re-created the release tag because of 1 issue. But seems the TF registry did not catch it up.
I am deleting the release tag and hoping the v1.2.2 information goes away from the registry versions. Then I will push the tag again and hope registry picks up the new version again.

Thanks for your patience.

**Comment by yussufsh** [comment]
Even after deleting the release the registry is not removing the version from their listing. I have emailed Terraform registry team with the issue asked them to remove v1.2.2.

**Comment by mt-inside** [comment]
This is preventing me from installing <2.0.0, which is preventing my entire flatcar linux fleet from booting, because flatcar is stuck on ignition 0.34...

From `TF_LOG=DEBUG terraform init`
it seems like it's actually looking at github for the checksums? Maybe that's what you meant.

I tried using the 1.2.1 tag from 2019 but it's only a tag not a Release, and it doesn't have the SHA256SUMS files.

```
2021/03/08 19:52:53 [DEBUG] GET https://registry.terraform.io/v1/providers/community-terraform-providers/ignition/1.2.2/download/linux/amd64                                                 
2021/03/08 19:52:53 [DEBUG] GET https://github.com/community-terraform-providers/terraform-provider-ignition/releases/download/v1.2.2/terraform-provider-ignition_1.2.2_SHA256SUMS        
```

For anyone else: I managed to make this work by building v1.2.1 locally (there's no v.1.2.2 tag?).
You then have to make terraform use it and NOT try to read any metadata from the registry.
This isn't easy at all. `terraform init -verify-plugins=false` seems to be ignored.

I ended up making a tfrc file in with the following blocks in it, and exporting that into `TF_CLI_CONFIG_FILE` in my shell.

The documentation for this is buried: https://www.terraform.io/docs/cli/config/config-file.html#development-overrides-for-provider-developers

**Comment by yussufsh** [comment]
The release 1.2.1 is with the upstream repository that you can use. We have not released those versions but the tags are listed from the upstream.

We have released v1.2.2 (and then re-released). The Terraform registry values were not updated with the latest information. I have written to the Hashicorp registry support but all I have till now is an automated reply that my ticket is raised.

**Comment by yussufsh** [comment]
I am thinking we can make a release called v1.3.0 which will be part of community-terraform-providers and v1 branch. This way we don't depend on Terraform registry support to delete 1.2.2.

We can work with v1.3.0 which is at same code level as 1.2.2.

**Comment by bgilbert** [comment]
Why not call it 1.2.3 in that case?  There wouldn't be any semantic changes in the new release.

**Comment by yussufsh** [comment]
Lets relate 1.3 as community-terraform-providers release and 1.2 as hashicorp legacy.

I have already released 1.3.0 now... any changes to v1 branch will have incremental releases to this version.


```
# cat main.tf
terraform {
  required_providers {
    ignition = {
      source  = "community-terraform-providers/ignition"
      version = "1.3.0"
    }
  }
}

# terraform init

Initializing the backend...

Initializing provider plugins...
- Finding community-terraform-providers/ignition versions matching "1.3.0"...
- Installing community-terraform-providers/ignition v1.3.0...
- Installed community-terraform-providers/ignition v1.3.0 (self-signed, key ID 13D373249FD8E4D3)

Partner and community providers are signed by their developers.
If you'd like to know more about provider signing, you can read about it here:
https://www.terraform.io/docs/plugins/signing.html

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

```
